tests: Add test for "autohide whiteboard toolbar"

pinhaum opened the pull request:
### What does this PR do?
This PR implements test coverage for the auto hide whiteboard toolbar feature in BigBlueButton. The test validates the functionality that allows users to enable/disable automatic hiding of the whiteboard toolbar through the application settings.

#### Changes Made
- Added autoHideWhiteboardToolbar() method in bigbluebutton-tests/playwright/options/options.js
- Added test case "Auto hide whiteboard toolbar" in bigbluebutton-tests/playwright/options/options.spec.js
- Added wbAutoHideToggleBtn selector in bigbluebutton-tests/playwright/core/elements.js

#### Notes
The test addresses the specific scenario mentioned in the issue, considering:
- Default state with whiteboardToolbarAutoHide: false setting
- User interaction to enable the auto-hide feature
- Dynamic toolbar visibility based on mouse position
- Settings persistence after modal confirmation

#### Video example


https://github.com/user-attachments/assets/49825212-9a57-48db-8631-f529fc0be2a2


### Closes Issue(s)
Closes #23797 

### How to test
1. run playwright options tests with:
`npm run test options` or `npx playwright test options`
you can also run the specific test using the command:
`npm run test:filter "Auto hide whiteboard toolbar"`
2. Confirm that the end meeting test passes and UI behaves as expected.

paultrudel committed:
Track presentation conversion in akka apps

paultrudel committed:
Add log for tracking conversion time

paultrudel committed:
Add meeting ID to conversion duration log

paultrudel committed:
Configure conversion timeout and wrap conversion map in scala class

paultrudel committed:
Add additional branch to match statement

paultrudel committed:
Enforce max conversion time for presentations

paultrudel committed:
Add specific message for conversion cancellation and fix breakout room issue

paultrudel committed:
Remove presentation documents after cancellation and add common message body

paultrudel committed:
Update log output

paultrudel committed:
retrigger checks

paultrudel committed:
Update conversion monitoring task

paultrudel committed:
Use blank slides after conversion cancellation

paultrudel committed:
Remove presentation conversion cancelled message

paultrudel committed:
Remove cancellation msg use in meeting actor

paultrudel committed:
Restrict all conversion timeouts based on page timeout

gustavotrott committed:
Update bbb-common-web/src/main/java/org/bigbluebutton/presentation/imp/SvgImageCreatorImp.java

gustavotrott committed:
Update bbb-common-web/src/main/java/org/bigbluebutton/presentation/imp/SvgImageCreatorImp.java

paultrudel committed:
Use correct pdffontstimeout

paultrudel committed:
Rerun ci tests

Arthurk12 committed:
fix: inconsistent hook component remounting due to count-based keys

In scenarios with sequential subscribe/unsubscribe events, the
subscription count may revert to the same value, causing the React
component key to remain unchanged and preventing remounting, which leads
to stale data and rendering inconsistencies in hook components and by
consequence in the hooks data provided to plugins. So this commit fixes
this by:
- Adding a 'version' field to data consumption hooks
- Incrementing version on each subscribe/unsubscribe event, independent of
  count
- Updating component keys to include version, forcing remounting when
  needed

gustavotrott committed:
metrics: Introduce traceLog to measure the timestamp that connectionAlive message cross each application of BBB

gustavotrott committed:
Add prometheus metrics for bbb_application_reach_latency_milliseconds

gustavotrott committed:
Add client logs about application RTT and error log if RTT is heigher than 2s

gustavotrott committed:
Improve client time sync with server and add clientSessionUUID as part of connection status table

Tainan404 committed:
Fix(html5): fix video stuttering on vimeo

Tainan404 committed:
Add more handles to promises

gustavotrott committed:
remove unnecessary log

gustavotrott committed:
small fixes

ramonlsouza committed:
enable quizzing by default

pinhaum committed:
test: add toggle auto hide toolbar option to elements list

pinhaum committed:
test: add test for auto hide whiteboard toolbar

ramonlsouza committed:
fix zoom position on slide change

antobinary committed:
build: Bump bbb-pads to 1.5.5

https://github.com/bigbluebutton/bbb-pads/releases/tag/v1.5.5

ramonlsouza committed:
fix inconsistent zoom

ramonlsouza committed:
skip whiteboard zoom test on CI - test needs to be updated

antobinary committed:
chore: npm dependencies bump

antobinary committed:
chore: Bump release to 3.0.14

gustavotrott committed:
ci: checkout from prebuilt merge ref

antonbsa commented on bigbluebutton-tests/playwright/options/options.js:
```diff
@@ -159,6 +159,29 @@ class Options extends MultiUsers {
+    await this.modPage.waitForSelector(e.whiteboard);
+    await this.modPage.hasElement(e.wbToolbar, 'should display the whiteboard toolbar');
+
+    const whiteboardLocator = await this.modPage.getLocator(e.whiteboard);
+    await expect(whiteboardLocator).toHaveScreenshot('whiteboard-with-toolbar-visible.png');
```
clear all notifications before the first screenshot to avoid unexpected elements appearing
<img width="377" height="180" alt="image" src="https://github.com/user-attachments/assets/8af633bb-d34c-4d07-be0b-659248143507" />

pinhaum replied:
validate to close all toasts notifications, will be commiting changes later

Tainan404 committed:
Fix(html5): Stale closure around timeSync

antonbsa commented on bigbluebutton-tests/playwright/options/options.js:
```diff
@@ -159,6 +159,29 @@ class Options extends MultiUsers {
+    await this.modPage.hasElement(e.wbToolbar, 'should display the whiteboard toolbar when hover the whiteboard');
+
+    await this.modPage.hoverElement(e.chatButton)
+    await sleep(1000);
```
instead of using a static timeout, prefer to check for an element change. using `toBeHidden` assertion or similar in this case does not work (i've already tried), but checking for a css style change should work:

<img width="1610" height="853" alt="image" src="https://github.com/user-attachments/assets/8bd780fa-fdcd-430b-949e-57e06983b6b3" />

pinhaum replied:
validated to check the css changes, will commit changes later

gustavotrott committed:
gql-server: tweak name of Type breakoutRoom_user_summary

gustavotrott committed:
fix: only moderators should see last breakout of users

gustavotrott committed:
add rule for bbb_client_not_in_meeting

antobinary committed:
docs: add link to BBB 3.0.14

ramonlsouza committed:
fix prop type

ramonlsouza committed:
fix finddomnode warning

hiroshisuga committed:
[3.0 external video share] Add stop share video button (#23786)

* Update component.tsx

* Update styles.ts

* Update en.json

* z-index:1 to make sure

* remove redundant role

* z-index:100

following the coderabbit, not sure if it makes sense...

* revert to z: 1

z:100 seems to have no effect.

* change case

* ignore ts warning on css file

---------

Co-authored-by: Ramón Souza <contato@ramonsouza.com>

JoVictorNunes committed:
fix(LAD): Fix unsafe optional chaining (#23833)

* fix(LAD): Fix unsafe optional chaining

* Changes by code review

ramonlsouza committed:
adjust textarea css selector to only target whiteboard (#23836)

antonbsa committed:
tests: Add support for SDK flaky test exclusions in CI workflow (#23656)

* tests: add support for SDK flaky test exclusions in CI workflow

* Update .github/workflows/automated-tests.yml

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

---------

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

transifex-integration[bot] committed:
Translate en.json in el_GR

100% translated source file: 'en.json'
on 'el_GR'.

pinhaum committed:
test: refac to:
- close all notifications before taking screenshot
- check for class elements changes when hiding the toolbar

pinhaum committed:
Merge remote-tracking branch 'upstream/v3.0.x-release' into test-autohide-toolbar

pinhaum commented:
Closed in favor of #23891
